src/vos/vos_aggregate.c

```diff
@@ -1491,12 +1491,6 @@ insert_segments(daos_handle_t ih, struct agg_merge_window *mw, bool last, unsign
 		D_AGG_ASSERTF(mw, rect.rc_ex.ex_lo <= rect.rc_ex.ex_hi,
 			      "phy_ent " DF_RECT " off=" DF_X64 "\n", DP_RECT(&phy_ent->pe_rect),
 			      phy_ent->pe_off);
-		D_AGG_ASSERTF(mw, phy_ent->pe_remove || rect.rc_ex.ex_lo <= mw->mw_ext.ex_hi,
-			      "phy_ent->pe_remove=%d phy_ent->pe_off=" DF_X64 " rect=" DF_RECT
-			      " mw=" DF_EXT "\n",
-			      phy_ent->pe_remove, phy_ent->pe_off, DP_RECT(&rect),
-			      DP_EXT(&mw->mw_ext));
-
```

Comment:
You finally figure out why it's incorrect? Maybe you'd explain it a bit in commit comment?

Reply:
This is a fairly minimal sequence that fails 

-w key@0-3  -w key@2-4 -R key@2-3 -w key@5-4 -R key@5-3 -a

Reply:
That one does trigger the assert.   My question is why does it trigger the assert as I think the window really should be flushed after the first entry.

Reply:
Oh, I read your message wrong, I was thinking you said it doesn't trigger the assert.

Merge window flush only happens in two scenarios so far: 1. when trying to enqueue a visible recx which is disjoint with merge window; 2. on last logical recx enqueue.

So the merge window flush for the two examples both happened when trying to enqueue the visible 5-1 recx.